src/bin/bat/app.rs

```diff
@@ -136,6 +136,12 @@ impl App {
             }
         });
 
+        if self.matches.value_of("file-name").is_some()
+            && self.matches.values_of("file-name").unwrap().len() != files.len()
```

Comment:
Instead of first checking for `.is_some()` and then using `.unwrap()` later, you can directly match on the `Option`:
``` rust
match self.matches.value_of("file-name") {
    Some(filenames) if filenames.len() != files.len => { … },
    _ => {}
}
```

Reply:
nice! The match statement continues to surprise me in its versatility.